### ISSUE TITLE
Training won't start

### ISSUE THREAD

**Issue by ardaboluk** [yhenon/pytorch-retinanet#201]
Stuck at loading training data for past 36 hrs. Is this normal, or am I missing something ?

CUDA available: True
loading annotations into memory...
Done (t=13.07s)
creating index...
index created!
loading annotations into memory...
Done (t=0.39s)
creating index...
index created!
Num training images: 118287

**Comment by Pratipkhandelwal** [comment]
Ideally, training should start just after the Num training images part.

Example of my Model Training :
CUDA available: True
No validation annotations provided.
Downloading: "https://download.pytorch.org/models/resnet50-19c8e357.pth" to ./resnet50-19c8e357.pth
100%|███████████████████████████████████████| 97.8M/97.8M [00:00<00:00, 156MB/s]
Num training images: 47
Epoch: 0 | Iteration: 0 | Classification loss: 1.12902 | Regression loss: 1.11606 | Running loss: 2.24508
Epoch: 0 | Iteration: 1 | Classification loss: 1.11770 | Regression loss: 1.09277 | Running loss: 2.22778
............

**Comment by ardaboluk** [comment]
If I change num_workers to 0 in train.py, training starts normally. Checked the backtrace from Pdb. It seems like it's about multiprocessing:

arda@arda:/home/arda/Projects/MOD_arda/tryit/src/pytorch-retinanet$ python -m pdb train.py --dataset coco --coco_path /home/arda/Projects/MOD_arda/mobilenet_convlstm_retinanet/coco2017 --depth 18
> /home/arda/Projects/MOD_arda/tryit/src/pytorch-retinanet/train.py(1)<module>()
-> import argparse
(Pdb) c
CUDA available: True
loading annotations into memory...
Done (t=13.95s)
creating index...
index created!
loading annotations into memory...
Done (t=0.48s)
creating index...
index created!
Num training images: 118287
^C
Program interrupted. (Use 'cont' to resume).
> /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/selectors.py(379)select()
-> for fd, event in fd_event_list:
(Pdb) bt
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/runpy.py(193)_run_module_as_main()
-> "__main__", mod_spec)
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/runpy.py(85)_run_code()
-> exec(code, run_globals)
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/pdb.py(1694)<module>()
-> pdb.main()
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/pdb.py(1667)main()
-> pdb._runscript(mainpyfile)
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/pdb.py(1548)_runscript()
-> self.run(statement)
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/bdb.py(434)run()
-> exec(cmd, globals, locals)
  <string>(1)<module>()
  /home/arda/Projects/MOD_arda/tryit/src/pytorch-retinanet/train.py(1)<module>()
-> import argparse
  /home/arda/Projects/MOD_arda/tryit/src/pytorch-retinanet/train.py(121)main()
-> for iter_num, data in enumerate(dataloader_train):
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/site-packages/torch/utils/data/dataloader.py(363)__next__()
-> data = self._next_data()
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/site-packages/torch/utils/data/dataloader.py(974)_next_data()
-> idx, data = self._get_data()
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/site-packages/torch/utils/data/dataloader.py(941)_get_data()
-> success, data = self._try_get_data()
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/site-packages/torch/utils/data/dataloader.py(779)_try_get_data()
-> data = self._data_queue.get(timeout=timeout)
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/multiprocessing/queues.py(104)get()
-> if not self._poll(timeout):
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/multiprocessing/connection.py(257)poll()
-> return self._poll(timeout)
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/multiprocessing/connection.py(414)_poll()
-> r = wait([self], timeout)
  /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/multiprocessing/connection.py(911)wait()
-> ready = selector.select(timeout)
> /home/arda/arda_workspace/anaconda3/envs/ardaenv/lib/python3.6/selectors.py(379)select()
-> for fd, event in fd_event_list:

**Comment by ardaboluk** [comment]
I'm closing the issue as the problem seems to be about Python multiprocessing and/or PyTorch.
